pydocx/export/base.py

```diff
@@ -180,6 +180,8 @@ def _convert_complex_fields_into_simple_fields(self):
                     previous_parent_new_children.append(child)
             first_run.parent.children = previous_parent_new_children
 
+            field.parent = first_run.parent
```

Comment:
The problem was field.parent was never being set, so it was None. Yet the new field was being inserted into the structure. As a consequence, it mean that the contained runs had no way to detect whether they were contained within a paragraph that was a heading, since the structure only went up as far as the field.